Deeploy/CommonExtensions/CodeTransformationPasses/MemoryAllocation.py

```
@@ -119,15 +255,16 @@ def apply(self,
         inputs = [buff for buff in memoryLevelBuffers if self.is_final_input(buff, name)]
 
         # We have to allocate the output buffers, unless they are global
+        # Topological sorting is necessary to ensure that we allocate reference buffers before their dependents
         for buffer in reversed(self.topologicallySortBuffers(outputs + transients)):
             assert buffer._live == False, f"Tried to allocate already live buffer {buffer.name}"
```

Ruff (0.14.14)

[error] Avoid equality comparisons to `False`; use `not buffer._live:` for false checks

Replace with `not buffer._live`

(E712)